src/Uno.UI/UI/Xaml/Shapes/Shape.cs

```diff
@@ -168,8 +168,13 @@ public DoubleCollection StrokeDashArray
 		);
 		#endregion
 
+		// Do not invoke base.IsViewHit(): We don't have to have de FrameworkElement.Background to be hit testable!
 		internal override bool IsViewHit()
-			=> Fill != null; // Do not invoke base.IsViewHit(): We don't have to have de FrameworkElement.Background to be hit testable!
+			=> Fill != null
+#if __SKIA__ // we only add this condition for skia because it's the only platform with proper hit-testing support for shapes. If we add it for other platforms, we get a different but still inaccurate behaviour, so we prefer to keep the behaviour as is.
+				|| Stroke != null
```

Comment:
FYI @Youssef1313 :)

Reply:
Great. That's the same change we'll need for Wasm as well so it will be `#if __SKIA__ || __WASM__` once **both** PRs are in.